packages/query-core/src/types.ts

```diff
@@ -43,8 +43,10 @@ export type DefaultError = Register extends {
 export type QueryKey = ReadonlyArray<unknown>
 
 export declare const dataTagSymbol: unique symbol
-export type DataTag<TType, TValue> = TType & {
+export declare const dataTagErrorSymbol: unique symbol
+export type DataTag<TType, TValue, TError = unknown> = TType & {
```

Comment:
```suggestion
export type DataTag<TType, TValue, TError = DefaultError> = TType & {
```

Reply:
So as per the discussion elsewhere - if DataTag is set without the 3rd argument, and a user sets TError via the QueryClient method, should this default win or should the TError win? I believe the second as the first could be a breaking change

Maybe we use a default of UnsetMarker here?

Reply:
aah I get it now. Yeah a separate marker might be more correct because you can set `DefaultError` globally to `unknown` as well and then want to use `TError` on the querfClient method and then it wouldn’t work I think

Reply:
Cool, already pushed the change so see what you make of it